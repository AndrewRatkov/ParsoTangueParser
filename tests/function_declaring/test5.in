def str something(int x, str y)
if x > 1 then
str z := y * 2;
return z + "aba";
fi
something(1, "cucu");
return "abacaba";
enddef

