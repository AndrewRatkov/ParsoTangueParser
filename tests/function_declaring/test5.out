[DEF FUN something(INTEGER x, STRING y)=>STR]
|   
+-->IF
|   |
|   +--->[INT,> ]------->[INT,x]
|   |        |
|   |        +---------->[INT,1]
|   |
|   THEN
|   |
|   +--->[DEF str z]--->[STR,* ]------->[STR,y]
|   |                       |
|   |                       +---------->[INT,2]
|   |
|   +--->[RETURN]--->[STR,+ ]------->[STR,z]
|   |                    |
|   |                    +---------->[STR,"aba"]
|   |
|   FI
|   
+-->[CALL STR something]------->[INT,1]
|             |
|             +---------------->[STR,"cucu"]
|   
+-->[RETURN]--->[STR,"abacaba"]
